### DISCUSSION TITLE
Fluentui TextField onChange not firing with yup schema validation

### DISCUSSION THREAD

**Discussion by vinothvenkatachalam** [react-hook-form/react-hook-form#7774] [Q&A] [answered]
### Version Number

 "react-hook-form": "^7.25.3"

### Codesandbox/Expo snack

https://codesandbox.io/s/sharp-sun-4hmyg?file=/src/App.js

### Steps to reproduce

1. Select comments text filed.
2. Enter comments and not able enter comments.

### Expected behaviour

Comments to be entered and TextField onChange needs to be fire.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Moshyfawn** [comment]
Your CSB is empty
<img width="1097" alt="image" src="https://user-images.githubusercontent.com/16290753/152804509-4f8dfc61-7456-421b-8548-ea614258f8f4.png">

**Comment by vinothvenkatachalam** [comment]
Hi,
Please find the below code snippet,

import React, { useState} from "react";
import ReactDOM from "react-dom";
import { useForm, Controller  } from "react-hook-form";
import { yupResolver } from "@hookform/resolvers/yup";
import * as yup from "yup";
import {
    TextField
} from '@fluentui/react';


const SignupSchema = yup.object().shape({
  firstName: yup.string().required(),
  age: yup.number().required().positive().integer(),
  website: yup.string().url(),
  comments:yup.string().required('Pls give comments')
});

export default function App() {
    const [textValue, setTextValue] = useState('');

  const {
    register,
    handleSubmit,
    formState: { errors }
  } = useForm({
    resolver: yupResolver(SignupSchema)
  });
  const onSubmit = (data) => {
    alert(JSON.stringify(data));
  };

  const textOnChange = (ev, newText) => setTextValue(newText);

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <div>
        <label>First Name</label>
        <input {...register("firstName")} />
        {errors.firstName && <p>{errors.firstName.message}</p>}
      </div>
      <div style={{ marginBottom: 10 }}>
        <label>Last Name</label>
        <input {...register("lastName")} />
        {errors.lastName && <p>{errors.lastName.message}</p>}
      </div>
      <div>
        <label>Age</label>
        <input type="number" {...register("age", { valueAsNumber: true })} />
        {errors.age && <p>{errors.age.message}</p>}
      </div>
      <div>
        <label>Website</label>
        <input {...register("website")} />
        {errors.website && <p>{errors.website.message}</p>}
      </div>
    <div>
        
    <TextField label="Comments"  value={textValue} onChange={textOnChange}
                            multiline rows={6} {...register("comments")}/>
    {errors.comments && (<p>{errors.comments.message}</p>)}
    </div>

      <input type="submit" />
    </form>
  );
}

**Comment by Moshyfawn** [comment]
Please, create a reproducible codesandbox next time instead of just dumping a plain code without even formatting.

I've copy-pasted you code-snippet into [this CSB](https://codesandbox.io/s/adoring-moser-ozop2?file=/src/App.js) and removed everything except the input that you've described having the issue with. It works for me: I'm able to enter a comment into the TextField input and submit the form with it's value.

Please, review [my CSB](https://codesandbox.io/s/adoring-moser-ozop2?file=/src/App.js) and edit it to match the behavior you've described in your issue description.

**Comment by vinothvenkatachalam** [comment]
Please don't remove onchange event of the TextField and try it. Bcoz onchange event i need to setTextValue.

the expectation is to fire onChange.

**Comment by Moshyfawn** [comment, marked as answer]
The `register` form method is used to work with **uncontrolled** inputs. By using some external state via `useState` and passing it into your TextField into it's `value`, you're making the TextField **controlled**, which `register` doesn't support.

There're 2 things you can do:
- **either** don't pass your `textValue` into the TextField's `value` and pass your `textOnChange` setter into `register`'s onChange callback to prevent the TextField's `onChange` callback from being overridden by `register` one.
```js
<TextField
  label="Comments"
  multiline
  rows={6}
  {...register("comments", {
    onChange: (event) => {
      textOnChange(event.target.value);
    }
  })}
/>
```
- **or** use `Controller` / `useController` to use your TextField as a **controlled** input and pass your `textOnChange` and `field.onChange` from the `Controller` / `useController` into the TextField's `onChange` callback
[Refer to RHF docs on `Controller`](https://react-hook-form.com/api/usecontroller/controller) for more info about it's use
